### PR TITLE
server/proxy: Add multimon support

### PR DESCRIPTION
**Pull request by kubistika**
This PR adds basic multimon support. Note that we do not forward the server's MonitorLayoutPDU yet.

### CODE REVIEW

File: server/proxy/pf_context.c [hunk 1]
```diff
@@ -77,7 +77,49 @@ rdpContext* p_client_context_create(rdpSettings* clientSettings,
 	settings->ServerPort = port;
 	settings->SoftwareGdi = FALSE;
 	settings->RedirectClipboard = FALSE;
+
+	/* Client Monitor Data */
+	settings->MonitorCount = clientSettings->MonitorCount;
```
Comment (speidy) [inline server/proxy/pf_context.c:82]: maybe wrap in an inline function, this func becomes a mess

**Comment by speidy** [comment]
we should sync Monitor Layout Pdu

**Comment by speidy** [comment]
what about syncing the client and server?

File: server/proxy/pf_context.c (hunk 1 above)
Comment (kubistika) [reply server/proxy/pf_context.c:82]: we need to figure out a way to use freerdp_settings_clone

File: include/freerdp/display.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * Input Interface API
```
Comment (speidy) [inline include/freerdp/display.h]: not input

File: libfreerdp/core/display.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * Activation Sequence
```
Comment (speidy) [inline libfreerdp/core/display.h]: display

File: libfreerdp/core/display.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+#ifndef FREERDP_LIB_CORE_DISPLAY_H
+#define FREERDP_LIB_CORE_DISPLAY_H
+
+#include "rdp.h"
+
+#include <freerdp/api.h>
+#include <freerdp/settings.h>
+
+BOOL freerdp_display_send_monitor_layout(rdpContext* context);
```
Comment (speidy) [inline libfreerdp/core/display.h]: should be `FREERDP_LOCAL BOOL display_send_monitor_layout()`

File: libfreerdp/core/display.c [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+		return FALSE;
+
+	if (!rdp_write_monitor_layout_pdu(st, settings->MonitorCount, settings->MonitorDefArray))
+	{
+		Stream_Release(st);
+		return FALSE;
+	}
+
+	return rdp_send_data_pdu(rdp, st, DATA_PDU_TYPE_MONITOR_LAYOUT, 0);
+}
```
Comment (speidy) [inline libfreerdp/core/display.c]: newline

File: include/freerdp/display.h [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+#ifndef FREERDP_DISPLAY_H
+#define FREERDP_DISPLAY_H
+
+#include <freerdp/freerdp.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+FREERDP_API BOOL freerdp_display_send_monitor_layout(rdpContext* input);
```
Comment (speidy) [inline include/freerdp/display.h]: input -> ?!

File: server/proxy/pf_client.c [hunk 7]
```diff
@@ -26,6 +26,7 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
+#include <freerdp/display.h>
```
Comment (speidy) [inline server/proxy/pf_client.c]: move down

File: server/proxy/pf_client.c [hunk 8]
```diff
@@ -90,6 +91,15 @@ static BOOL pf_client_end_paint(rdpContext* context)
 	return ps->update->EndPaint(ps);
 }
 
+static BOOL pf_client_remote_monitors(rdpContext* context, UINT32 count, const MONITOR_DEF* monitors)
+{
+	pClientContext* pc = (pClientContext*) context;
+	proxyData* pdata = pc->pdata;
+	rdpContext* ps = (rdpContext*)pdata->ps;
+
+	return freerdp_display_send_monitor_layout(ps);
```
Comment (speidy) [inline server/proxy/pf_client.c:100]: We don’t pass the monitors we got at all :(((

Comment (speidy) [reply server/proxy/pf_client.c:100]: We might want to update the settings as well

Comment (kubistika) [reply server/proxy/pf_client.c:100]: lol i'll fix it sorry

**Comment by kubistika** [comment]
irrelevant, see #19

**Pull request closed without merging**